zul/src/archive/web/js/zul/wgt/Toolbar.js

```diff
@@ -50,7 +50,144 @@ zul.wgt.Toolbar = zk.$extends(zul.Widget, {
+		 *
+		 * @param boolean overflowPopup whether toolbar has a button that shows a popup
+		 * @since 8.6.0
+		 */
+		overflowPopup: function () {
+			this.rerender();
+		}
+	},
+	bind_: function () {
+		this.$supers('bind_', arguments);
+		if (this.isOverflowPopup()) {
+			zWatch.listen({onFloatUp: this, onCommandReady: this, onSize: this});
+			this.domListen_(this.$n('overflowpopup-button'), 'onClick', '_openPopup');
+		}
+	},
+	unbind_: function () {
+		var popup = this.$n('popup');
+		if (popup) {
+			this.domUnlisten_(this.$n('overflowpopup-button'), 'onClick', '_openPopup');
+			zWatch.unlisten({onFloatUp: this, onCommandReady: this, onSize: this});
+		}
+		this.$supers('unbind_', arguments);
+	},
+	_openPopup: function (evt) {
+		if (this._open)
+			return;
+
+		this._open = true;
+		var popup = this.$n('popup');
+		this.setFloating_(true, {node: popup});
+		zWatch.fire('onFloatUp', this);
+		var topZIndex = this.setTopmost();
+		popup.style.zIndex = topZIndex > 0 ? topZIndex : 1;
+		jq(popup).removeClass(this.$s('popup-close')).addClass(this.$s('popup-open')).zk.makeVParent();
+		this._syncPopupPosition();
+	},
+	_syncPopupPosition: function () {
+		zk(this.$n('popup')).position(this.$n(), 'after_end');
+	},
+	onFloatUp: function (ctl) {
+		if (!zUtl.isAncestor(this, ctl.origin))
+			this._closePopup();
+	},
+	_closePopup: function () {
+		if (!this._open)
+			return;
+
+		this._open = false;
+		var jqPopup = jq(this.$n('popup'));
+		jqPopup.removeClass(this.$s('popup-open')).addClass(this.$s('popup-close')).zk.undoVParent();
+		this.setFloating_(false);
+	},
+	onCommandReady: function () {
+		if (this.desktop && this.isOverflowPopup())
+			this._adjustContent();
+	},
+	onSize: function () {
+		this.$supers('onSize', arguments);
+		if (this.desktop && this.isOverflowPopup()) {
+			this._adjustContent();
+			if (this._open)
+				this._syncPopupPosition();
+		}
+	},
+	_adjustContent: function () {
+		if (zUtl.isImageLoading()) {
+			setTimeout(this.proxy(this._adjustContent), 20);
+			return;
+		}
+
+		var jqToolbar = jq(this),
+			contentWidth = jqToolbar.width() - jq(this.$n('overflowpopup-button')).width(),
+			popup = this.$n('popup'),
+			cave = this.$n('cave'),
+			oldToolbarChildren = jq(cave).children().toArray(),
+			oldPopupChildren = jq(popup).children().toArray(),
+			children = oldToolbarChildren.concat(oldPopupChildren),
+			childrenAmount = children.length,
+			tempChildrenWidth = 0,
+			newPopupChildrenAmount = 0;
+
+		// Calculate width to decide how many children should be displayed on popup.
+		for (var i = 0; i < childrenAmount; i++) {
+			tempChildrenWidth += jq(children[i]).outerWidth(true);
+			if (tempChildrenWidth >= contentWidth) {
+				newPopupChildrenAmount = childrenAmount - i;
+				break;
+			}
+		}
+		// If overflow popup has more than 8 children,
+		// setting its height as the sum of the top eight popup children's heights to enable scroll bar.
+		if (newPopupChildrenAmount > 8) {
```

Comment:
What if the the first children is `<div height="1000px;" />`?

Perhaps we need a fixed height (250 / 350?).
https://developer.mozilla.org/en-US/docs/Web/CSS/max-height